localstack/utils/aws/message_forwarding.py

```diff
@@ -39,6 +39,7 @@ def send_event_to_target(
     source_service: str = None,
 ):
     region = extract_region_from_arn(target_arn)
+    account_id = extract_account_id_from_arn(target_arn)
```

Comment:
We should use the source arn here conceptually - not sure if cross account is allowed here, but if it is, the source arn would be the way to go. Services that do not allow cross account requests (like firehose) would not work, but that would be fine.